include/deal.II/fe/fe_interface_values.h

```diff
@@ -25,6 +25,342 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+#ifndef DOXYGEN
+template <int dim, int spacedim>
+class FEInterfaceValues;
+#endif
+
+/**
+ * Namespace for views you get from accessing FEInterfaceValues using an
+ * extractor.
+ */
+namespace FEInterfaceViews
+{
+  /**
+   * The base class for the views.
+   */
+  template <int dim, int spacedim = dim>
+  class Base
+  {
+  public:
+    using value_type = double;
+
+    Base(const FEInterfaceValues<dim, spacedim> &fe_interface);
+
+  protected:
+    /**
+     * Store a pointer to the FEInterfaceValues instance.
+     */
+    const FEInterfaceValues<dim, spacedim> *fe_interface;
+  };
+
+
+
+  /**
+   * The view of a scalar variable for FEInterfaceValues.
+   */
+  template <int dim, int spacedim = dim>
+  class Scalar : public Base<dim, spacedim>
+  {
+  public:
+    /**
+     * This is the type returned for gradients, for example from
+     * average_gradient().
+     */
+    using gradient_type = dealii::Tensor<1, spacedim>;
```

Comment:
I think it would actually be nice if we could inherit this information from the place where it really comes from:
```suggestion
    using gradient_type = typename FEValuesViews::Scalar<dim,spacedim>::gradient_type;
```
This way we don't have to duplicate the information.